lib/zlib.js

```diff
@@ -114,7 +119,7 @@ function zlibBuffer(engine, buffer, callback) {
     var err = null;
 
     if (nread >= kMaxLength) {
-      err = new RangeError(kRangeErrorMessage);
+      err = new errors.RangeError('ERR_INVALID_RANGE', kRangeErrorMessage);
```

Comment:
I created **ERR_INVALID_RANGE** like a generic range error message.

`const kRangeErrorMessage = 'Cannot create final Buffer. It would be larger than 0x${kMaxLength.toString(16)} bytes;`